### DISCUSSION TITLE
Client/Server component  vs Client Side/Server Side/Static Generation

### DISCUSSION THREAD

**Discussion by kokushkin** [vercel/next.js#50790] [Help]
Hi! I've been doing the tutorial and reading the beginning of the docs (React Essentials), and I'm having a little  problem trying to understand how these concepts related in nextjs.  

![zoq4gl2c.png](https://github.com/vercel/next.js/assets/9853656/2745f204-d6cf-4508-a1a3-1aef78198e05)

Particticury,  is it possible to do Server Side Renderjng of a Client Component? 

Does it have something to do with App Router vs Pages Router?  I noticed that  tutorial only written for Pages  Router, but the documentation has both versions. Are all of these concepts still existing in App Router?

**Comment by icyJoseph** [comment]
In Next.js you are given the advantage that client components, are rendered server side, for pre-rendering purpose, though if a suspense boundary is hit the fallback might be rendered instead.

The React Essentials section also say, https://nextjs.org/docs/getting-started/react-essentials#client-components:

> Client Components enable you to add client-side interactivity to your application. In Next.js, they are [pre-rendered](https://nextjs.org/docs/app/building-your-application/rendering#component-level-client-and-server-rendering) on the server and [hydrated](https://nextjs.org/docs/app/building-your-application/rendering#component-level-client-and-server-rendering) on the client. You can think of Client Components as how components in the [Pages Router](https://nextjs.org/docs/pages) have always worked.

> Components in the Client Component module graph are primarily rendered on the client, but with Next.js, they can also be pre-rendered on the server and hydrated on the client.

In general, Next.js will always try to render in such a way that doesn't hurt SEO. So it is a good idea to not make any assumptions about window/document (DOM) presence at all.

- https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#seo
